apollo-router/src/plugin/mod.rs

```diff
@@ -187,8 +187,36 @@ impl PluginFactory {
     pub(crate) fn is_apollo(&self) -> bool {
         self.name.starts_with("apollo.") || self.name.starts_with("experimental.")
     }
+
+    /// Create a plugin factory.
+    pub fn new<P: PluginUnstable>(group: &str, name: &str) -> PluginFactory {
```

Comment:
Perhaps better named as `new_unstable` ?

Reply:
What's the reason we need `new` and `new_private` at all?

Reply:
it is used in the `register_plugin` macro